js/token-tree.js

```diff
@@ -36,9 +47,35 @@
    */
   function selectEventHandler (event) {
     if (!event.choice.children) {
+      insertTokenIntoTextBox.call(this, event);
+
       return;
     }
+    toggleTreeElement(event);
+  }
+
+  /**
+   * @param {object} event event object
+   */
+  function insertTokenIntoTextBox (event) {
+    var token = event.choice.id;
+    var field = $(this).data('field');
+    if (field.indexOf('html') < 0) {
+      field = textMsgID($(this));
```

Comment:
this is a bit confusing. What does the `textMsgID` function do? Can we use a better name?

Reply:
These are all copied from the Core file, so I feel its better not to change stuff, as it will be confusing.